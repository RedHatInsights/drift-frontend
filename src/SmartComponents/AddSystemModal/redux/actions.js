import types from './types';

function toggleAddSystemModal() {
    return {
        type: types.OPEN_ADD_SYSTEM_MODAL
    };
}

function selectActiveTab(newActiveTab) {
    return {
        type: types.SELECT_ACTIVE_TAB,
        payload: newActiveTab
    };
}

function setSelectedSystemIds(selectedSystemIds) {
    return {
        type: types.SET_SELECTED_SYSTEMS_COMPARISON,
        payload: selectedSystemIds
    };
}

export default {
    toggleAddSystemModal,
    selectActiveTab,
    setSelectedSystemIds
};
