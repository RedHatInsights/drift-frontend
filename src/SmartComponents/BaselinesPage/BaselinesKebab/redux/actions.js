import types from './types';

function exportToCSV(baselineTableData) {
    return {
        type: types.EXPORT_BASELINES_LIST_TO_CSV,
        payload: baselineTableData
    };
}

export default {
    exportToCSV
};
