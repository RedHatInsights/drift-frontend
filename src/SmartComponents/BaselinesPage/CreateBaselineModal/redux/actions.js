import types from './types';

function toggleCreateBaselineModal() {
    return {
        type: types.TOGGLE_CREATE_BASELINE_MODAL
    };
}

export default {
    toggleCreateBaselineModal
};
