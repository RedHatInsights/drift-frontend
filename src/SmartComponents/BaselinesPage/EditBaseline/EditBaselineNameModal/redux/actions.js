import types from './types';

function toggleEditNameModal() {
    return {
        type: types.TOGGLE_EDIT_NAME_MODAL
    };
}

export default {
    toggleEditNameModal
};
