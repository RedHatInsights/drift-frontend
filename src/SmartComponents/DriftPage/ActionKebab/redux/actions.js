import types from './types';

function toggleKebab() {
    return {
        type: types.TOGGLE_KEBAB
    };
}

export default {
    toggleKebab
};
