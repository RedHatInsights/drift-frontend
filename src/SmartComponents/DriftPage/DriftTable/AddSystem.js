import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@patternfly/react-core';
import { AddCircleOIcon } from '@patternfly/react-icons';

export class AddSystem extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="add-system-header">
                <div className="add-system-icon">
                    <AddCircleOIcon/>
                </div>
                <Button
                    variant='primary'
                    onClick={ this.props.getAddSystemModal }>
                    Add System
                </Button>
            </div>
        );
    }
}

AddSystem.propTypes = {
    getAddSystemModal: PropTypes.func
};
