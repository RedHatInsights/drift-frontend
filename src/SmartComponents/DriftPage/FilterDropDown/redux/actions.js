import types from './types';

function toggleFilterDropDown() {
    return {
        type: types.OPEN_FILTER_DROPDOWN
    };
}

export default {
    toggleFilterDropDown
};
