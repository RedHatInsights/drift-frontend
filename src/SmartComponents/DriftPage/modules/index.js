export { default as DriftPageActions } from './actions';
export { default as DriftPageTypes } from './types';
export { default as DriftPageReducers } from './reducers';
