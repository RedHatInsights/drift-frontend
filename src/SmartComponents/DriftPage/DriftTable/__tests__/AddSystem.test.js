import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AddSystem } from '../AddSystem';
import { toggleAddSystemModal } from '../../../modules/actions';

describe('add system button', () => {
    it('should render correctly', async () => {
        const button = render(
            <AddSystem
                getAddSystemModal={ toggleAddSystemModal }
            />);
        expect(toJson(button)).toMatchSnapshot();
    });
});
