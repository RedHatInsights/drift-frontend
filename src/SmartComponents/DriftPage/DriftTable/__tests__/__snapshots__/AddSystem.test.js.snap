// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add system button should render correctly 1`] = `
<button
  class="pf-c-button pf-m-primary"
  type="button"
>
  Add System
</button>
`;
